### PR TITLE
test data generation

### PR DESCRIPTION
**Pull request by SantamRC**
@jpivarski check this out. 
In here, if we run `python dev/generate-strategies.py` it creates composite strategies by making use of contraints.json for particular kernel functions inside hypothesis-tests-spec folder. 
The plan is , after the strategies are generated, I would call those strategies and fetch the inputs they generate. For each of those input I would dynamically call the kernel and store the input and the output in a json file (tests-data/json). 
Then make a change in the current generate-tests.py to fetch those input and outputs and generate the test cases.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -14,6 +14,10 @@ tests-cuda-kernels/test*
 tests-spec/kernels.py
 tests-spec/test*
 
+##Kernel tests using hypothesis
+hypothesis-tests-spec/strategies.py
```
Comment (ianna) [inline .gitignore:18]: @SantamRC - is it something you wrote?

File: constraints.json [hunk 2]
```diff
@@ -0,0 +1,48 @@
+{
+    "kernels":[
+        {
+            "name":"awkward_BitMaskedArray_to_ByteMaskedArray",
```
Comment (ianna) [inline constraints.json:4]: I guess it's a test name. Are you planning on writing ~2000+ constraints for each test?

File: constraints.json [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+            "arguments":[
+                {
+                    "name":"BitMaskedArray-mask",
+                    "var":"mask",
```
Comment (ianna) [inline constraints.json:8]: I guess it's a variable name in this test. What if someone decides to name it differently?

File: constraints.json [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+                    "role":"Stores the mask array",
+                    "strategy_name":"lists",
+                    "constraints": {
+                        "min_size":0,
+                        "max_size":6,
```
Comment (ianna) [inline constraints.json:13]: what does it define? A maximum value or a maximum length?

**Review by ianna — COMMENTED** [review]
Good start. I couldn't run it due to missing "`strategies.py`" file.

**Comment by jpivarski** [comment]
@SantamRC Is this PR still active, or is it superseded by #971?

**Comment by SantamRC** [comment]
> @SantamRC Is this PR still active, or is it superseded by #971?

This PR is not active anymore

**Pull request closed without merging**